analyzers/tests/SonarAnalyzer.UnitTest/Wrappers/TypeInfoExtensionsTest.cs

```diff
@@ -0,0 +1,169 @@
+            nullability.Should().Be(new NullabilityInfo(NullableAnnotation.NotAnnotated, NullableFlowState.NotNull))
+                .And.BeEquivalentTo(new
+                {
+                    typeInfo.Nullability.Annotation,
+                    typeInfo.Nullability.FlowState
+                }, options => options.ComparingEnumsByName());
+
+            var convertedNullability = typeInfo.ConvertedNullability();
+            convertedNullability.Should().Be(new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.MaybeNull))
+                .And.BeEquivalentTo(new
+                {
+                    typeInfo.ConvertedNullability.Annotation,
+                    typeInfo.ConvertedNullability.FlowState
+                }, options => options.ComparingEnumsByName());
+        }
+
+        [TestMethod]
+        public void NullabilityInfoIsMissingIfNullableDisabled()
+        {
+            var code = @"
+#nullable disable
+public class C
+{
+    public void M()
+    {
+        object? o = null;
+        o.ToString();
+    }
+}
+";
+            var (tree, semanticModel) = TestHelper.CompileCS(code);
+            var identifier = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Last(x => x.NameIs("o")); // o in o.ToString()
+            var typeInfo = semanticModel.GetTypeInfo(identifier);
+            typeInfo.Nullability().Should().Be(new NullabilityInfo(NullableAnnotation.None, NullableFlowState.None));
+            typeInfo.ConvertedNullability().Should().Be(new NullabilityInfo(NullableAnnotation.None, NullableFlowState.None));
+        }
+
+        [DataTestMethod]
+        [DataRow("Debug.Assert(o != null);")]
+        [DataRow("Debug.Fail(string.Empty);")]
+        public void NullabilityInfoRecognizesDebug(string debug)
+        {
+            var code = $@"
+#nullable enable
+using System.Diagnostics;
+public class C
+{{
+    public void M()
+    {{
+        object? o = null;
+        {debug}
+        o.ToString();
+    }}
+}}
+";
+            var (tree, semanticModel) = TestHelper.CompileCS(code);
+            var identifier = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Last(x => x.NameIs("o")); // o in o.ToString()
+            var typeInfo = semanticModel.GetTypeInfo(identifier);
+            var expected =
+#if NET
+                new NullabilityInfo(NullableAnnotation.NotAnnotated, NullableFlowState.NotNull);
+#else
+                new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.MaybeNull);
+#endif
+            typeInfo.Nullability().Should().Be(expected);
+            typeInfo.ConvertedNullability().Should().Be(expected);
+        }
+
+        [DataTestMethod]
+        [DataRow(typeof(NullableAnnotation), typeof(Microsoft.CodeAnalysis.NullableAnnotation))]
+        [DataRow(typeof(NullableFlowState), typeof(Microsoft.CodeAnalysis.NullableFlowState))]
+        public void NullableEnumsAreIdentical(Type ours, Type theirs)
```

Comment:
Shim is not really ours :) 
```suggestion
        public void NullableEnumsAreIdentical(Type fromShim, Type fromRoslyn)
```